### ISSUE TITLE
Slider (QSlider) getvalue throws "Unable to get value"

### ISSUE THREAD

**Issue by Tieho** [ldtp/cobra#24]
Hello,
I have a Windows Qt application with a horizontal QSlider. I'm trying to interact with it using LDTP's getvalue, increase, decrease, etc functions. All these functions give me error "Unable to get value".

I have also push buttons and check boxes which I'm able to interact with using LDTP.

My Qt version is 5.6 beta, VS2015. Windows 10.

Here's the whole error:
Traceback (most recent call last):
  File "test.py", line 8, in <module>
    getvalue('MainWindow', 'sldr0')
  File "C:\Program Files (x86)\VMware\CobraWinLDTP\ldtp\client.py", line 72, in __call__
    return self.__send(self.__name, args)
  File "C:\Python27\lib\xmlrpclib.py", line 1599, in __request
    verbose=self.__verbose
  File "C:\Program Files (x86)\VMware\CobraWinLDTP\ldtp\client.py", line 193, in request
    raise LdtpExecutionError(e.faultString.encode('utf-8'))
ldtp.client_exception.LdtpExecutionError: Unable to get value

**Comment by nagappan** [comment]
I don't have Windows 10 environment, on Tuesday let me try to find Windows 10 image @work.
Try this:
From command line:
set LDTP_DEBUG=1
cobrawinldtp.exe

Now run your program and see whats printed on console ?

**Comment by MrTMan** [comment]
I ran my program via the LDTP script, and this was printed to the console (I believe sldr0 is the slider I'm trying to interact with):

Added: ControlType.Window : MainWindow : System.Int32[]
Window list count: 5
Processing request
Waiting for clients
Window: pane0 : \A(?ms)MainWindow\Z(?ms)
Window: frmCommandPrompt-cobrawinldtp.exe : \A(?ms)MainWindow\Z(?ms)
Window: frmQt5.632-bitforDesktop(MSVC2015)-pythontest.py : \A(?ms)MainWindow\Z(?ms)
Window: paneProgramManager : \A(?ms)MainWindow\Z(?ms)
Window: frmMainWindow : \A(?ms)MainWindow\Z(?ms)
MainWindow - Window found
object is non null: MainWindow
Processing request
Waiting for clients
Window: pane0 : \A(?ms)MainWindow\Z(?ms)
Window: frmCommandPrompt-cobrawinldtp.exe : \A(?ms)MainWindow\Z(?ms)
Window: frmQt5.632-bitforDesktop(MSVC2015)-pythontest.py : \A(?ms)MainWindow\Z(?ms)
Window: paneProgramManager : \A(?ms)MainWindow\Z(?ms)
Window: frmMainWindow : \A(?ms)MainWindow\Z(?ms)
MainWindow - Window found
object is non null: MainWindow
Current object Name is null
sldr0 : ukn0 :  : \A(?ms)sldr0\Z(?ms)
False : False
Obj name: System : ControlType.MenuBar
###mnuSystem###
sldr0 : mnuSystem : System : \A(?ms)sldr0\Z(?ms)
False : False
Obj name: System : ControlType.MenuItem
Menuitem shortcut length: 1
###mnuSystem###
sldr0 : mnuSystem1 : System : \A(?ms)sldr0\Z(?ms)
False : False
InternalGetObjectHandle: Child handle NULL
Obj name: Minimize : ControlType.Button
###btnMinimize###
sldr0 : btnMinimize : Minimize : \A(?ms)sldr0\Z(?ms)
False : False
InternalGetObjectHandle: Child handle NULL
Obj name: Maximize : ControlType.Button
###btnMaximize###
sldr0 : btnMaximize : Maximize : \A(?ms)sldr0\Z(?ms)
False : False
InternalGetObjectHandle: Child handle NULL
Obj name: Close : ControlType.Button
###btnClose###
sldr0 : btnClose : Close : \A(?ms)sldr0\Z(?ms)
False : False
InternalGetObjectHandle: Child handle NULL
Current object Name is null
sldr0 : ukn1 :  : \A(?ms)sldr0\Z(?ms)
False : False
Current object Name is null
sldr0 : ukn2 :  : \A(?ms)sldr0\Z(?ms)
False : False
Current object Name is null
sldr0 : ukn3 :  : \A(?ms)sldr0\Z(?ms)
False : False
InternalGetObjectHandle: Child handle NULL
Obj name: Nappi 1 : ControlType.Button
###btnNappi1###
sldr0 : btnNappi1 : Nappi1 : \A(?ms)sldr0\Z(?ms)
False : False
InternalGetObjectHandle: Child handle NULL
Obj name: Rasti 1 : ControlType.CheckBox
###chkRasti1###
sldr0 : chkRasti1 : Rasti1 : \A(?ms)sldr0\Z(?ms)
False : False
InternalGetObjectHandle: Child handle NULL
Current object Name is null
sldr0 : sldr0 :  : \A(?ms)sldr0\Z(?ms)
False : True
True : ControlType.Slider

**Comment by nagappan** [comment]
It has found the object, but unable to get the value

https://github.com/ldtp/cobra/blob/master/Ldtpd/Value.cs#L115

**Comment by Tieho** [comment]
This problem still persists, and also occurs on Windows 7. Any fix for this?

**Comment by Tieho** [comment]
I got GetValue working with ValuePattern instead of RangeValuePattern. Code snippet here:

```
if (childHandle.TryGetCurrentPattern(ValuePattern.Pattern,
    out pattern))
{
    if (((ValuePattern)pattern).Current.IsReadOnly)
    {
        throw new XmlRpcFaultException(123,
            "Control is read-only.");
    }
    return Convert.ToDouble(((ValuePattern)pattern).Current.Value, CultureInfo.InvariantCulture);
}
```

But I'm unable to making set value work (SetValue, Increase, Decrease) with this. There are no minimum and maximum values for ValuePattern, and even if I skip those, I get an error when trying to SetValue using ValuePattern. The error is:
`xmlrpclib.Fault: <Fault 0: 'Member not found. (Exception from HRESULT: 0x80020003 (DISP_E_MEMBERNOTFOUND))'>`

Qt version is 5.6.0.

**Comment by Tieho** [comment]
Get/Set value is fixed in commit 7c89385093cb63809ce274593510c3bc9ec21857.
Note: Qt change required for SetValue to work: https://codereview.qt-project.org/#/c/171450/
